### PR TITLE
fix(olm): IntegrationPlatform reconciled by operator 

### LINKED ISSUE #3161: Error when running camel-k in global mode with label selection
I've installed camel-k in global mode wit the following command:

```
kamel install --olm=false --namespace bar --global --operator-env-vars KAMEL_OPERATOR_ID=foo
```

Which generates a deployment with the following env vars:

```yaml
    spec:
      containers:
      - args:
        - --monitoring-port=8080
        - --health-port=8081
        command:
        - kamel
        - operator
        env:
        - name: WATCH_NAMESPACE
        - name: OPERATOR_NAME
          value: camel-k
        - name: POD_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.name
        - name: NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
        - name: KAMEL_OPERATOR_ID
          value: foo
```

At this point, if I create a kamelet like

```yaml
apiVersion: camel.apache.org/v1alpha1
kind: KameletBinding
metadata:
  annotations:
    camel.apache.org/operator.id: foo
    trait.camel.apache.org/container.image: acme/bz:0.1.17
  labels:
  name: c94kmpbc5fc71cmm8g6g
spec:
  sink:
    ref:
      apiVersion: camel.apache.org/v1alpha1
      kind: Kamelet
      name: og-sink
  source:
    ref:
      apiVersion: camel.apache.org/v1alpha1
      kind: Kamelet
      name: timer-source
```      

The operator fails with:

```json
{
    "level": "error",
    "ts": 1648975968.5594068,
    "logger": "controller.integration-controller",
    "msg": "Reconciler error",
    "reconciler group": "camel.apache.org",
    "reconciler kind": "Integration",
    "name": "mctr-c94kmpbc5fc71cmm8g6g",
    "namespace": "mctr-c947n1bc5fc71cmlhrdg",
    "error": "error during trait customization: unable to find catalog matching version requirement: runtime=, provider=quarkus",
    "errorVerbose": "unable to find catalog matching version requirement: runtime=, provider=quarkus\nerror during trait customization\ngithub.com/apache/camel-k/pkg/trait.Apply\n\tgithub.com/apache/camel-k/pkg/trait/trait.go:48\ngithub.com/apache/camel-k/pkg/controller/integration.(*platformSetupAction).Handle\n\tgithub.com/apache/camel-k/pkg/controller/integration/platform_setup.go:54\ngithub.com/apache/camel-k/pkg/controller/integration.(*reconcileIntegration).Reconcile\n\tgithub.com/apache/camel-k/pkg/controller/integration/integration_controller.go:309\ngithub.com/apache/camel-k/pkg/util/monitoring.(*instrumentedReconciler).Reconcile\n\tgithub.com/apache/camel-k/pkg/util/monitoring/controller.go:70\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.9.7/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.9.7/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.9.7/pkg/internal/controller/controller.go:214\nruntime.goexit\n\truntime/asm_amd64.s:1371",
    "stacktrace": "sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.9.7/pkg/internal/controller/controller.go:214"
}
```

### PR DESCRIPTION
**Pull request by squakez**
Closes #3161

**Release Note**
```release-note
fix(olm): IntegrationPlatform reconciled by operator 
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by squakez** [comment]
> The annotation should probably be transported also here:
> 
> https://github.com/apache/camel-k/blob/7942108bf0e00b322580387773eb7222bc6c0d18/pkg/trait/platform.go#L118

I've tried to cascade the annotation there, but I noticed that we cannot reach that point because we always get the IntegrationPlatform present in the operator namespace (see https://github.com/apache/camel-k/blob/7942108bf0e00b322580387773eb7222bc6c0d18/pkg/trait/platform.go#L104)

**Comment by astefanutti** [comment]
> > The annotation should probably be transported also here:
> > https://github.com/apache/camel-k/blob/7942108bf0e00b322580387773eb7222bc6c0d18/pkg/trait/platform.go#L118
> 
> I've tried to cascade the annotation there, but I noticed that we cannot reach that point because we always get the IntegrationPlatform present in the operator namespace (see
> 
> https://github.com/apache/camel-k/blob/7942108bf0e00b322580387773eb7222bc6c0d18/pkg/trait/platform.go#L104
> 
> )

Even when the operator is installed with OLM?

**Comment by squakez** [comment]
> Even when the operator is installed with OLM?

I haven't tested that scenario. Let me check how it is affected in such case.

File: e2e/common/cli/install_test.go [hunk 1]
```diff
@@ -24,27 +24,27 @@ package common
 
 import (
 	"bytes"
-	"github.com/apache/camel-k/pkg/install"
-	"github.com/apache/camel-k/pkg/util/defaults"
-	"github.com/apache/camel-k/pkg/util/kubernetes"
-	"github.com/apache/camel-k/pkg/util/openshift"
-	console "github.com/openshift/api/console/v1"
```
Comment (tadayosi) [inline e2e/common/cli/install_test.go:31]: looks like build error here

**Review by tadayosi — APPROVED** [review]

**Comment by squakez** [comment]
We'll need to include a proper E2E test for OLM installation once #3180 is ready.

File: e2e/common/cli/install_test.go (hunk 1 above)
Comment (squakez) [reply e2e/common/cli/install_test.go:31]: Good catch, thanks!

**Pull request merged**